### PR TITLE
refactor: added null check while searching for dashboard

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
Certain fields may have null values when generating a dashboard through JSON import. This pull request includes a check to exclude the search for null values in the dashboard.

### CODE REVIEW

File: frontend/src/container/ListOfDashboard/utils.ts [hunk 1]
```diff
@@ -3,18 +3,18 @@ import { Dashboard } from 'types/api/dashboard/getAll';
...
+): Dashboard[] => {
 	// Convert the searchValue to lowercase for case-insensitive search
 	const searchValueLowerCase = searchValue.toLowerCase();
 
 	// Use the filter method to find matching objects
 	return dashboardList.filter((item: Dashboard) => {
 		// Convert each property value to lowercase for case-insensitive search
 		const itemValues = Object.values(item?.data).map((value) =>
-			value.toString().toLowerCase(),
+			value?.toString().toLowerCase(),
```
Comment (Vikrant2520) [inline frontend/src/container/ListOfDashboard/utils.ts:14]: can we filter out the null values and then operate on the remaining data? protects us from adding null checks everywhere

Comment (Rajat-Dabade) [reply frontend/src/container/ListOfDashboard/utils.ts]: Done.

File: frontend/src/container/ListOfDashboard/utils.ts [hunk 2]
```diff
@@ -3,16 +3,16 @@ import { Dashboard } from 'types/api/dashboard/getAll';
...
-
 	// Use the filter method to find matching objects
 	return dashboardList.filter((item: Dashboard) => {
 		// Convert each property value to lowercase for case-insensitive search
-		const itemValues = Object.values(item?.data).map((value) =>
-			value.toString().toLowerCase(),
-		);
+		const itemValues = Object.values(item?.data).map((value) => {
+			if (value === null || value === undefined) return '';
+			return value?.toString().toLowerCase();
```
Comment (Vikrant2520) [inline frontend/src/container/ListOfDashboard/utils.ts:14]: null check won't be required in this line now

Comment (Rajat-Dabade) [reply frontend/src/container/ListOfDashboard/utils.ts]: Done.

**Review by Vikrant2520 — APPROVED** [review]
LGTM

File: frontend/src/container/ListOfDashboard/utils.ts [hunk 3]
```diff
@@ -3,16 +3,16 @@ import { Dashboard } from 'types/api/dashboard/getAll';
...
 	const searchValueLowerCase = searchValue.toLowerCase();
-
 	// Use the filter method to find matching objects
 	return dashboardList.filter((item: Dashboard) => {
 		// Convert each property value to lowercase for case-insensitive search
-		const itemValues = Object.values(item?.data).map((value) =>
-			value.toString().toLowerCase(),
-		);
+		const itemValues = Object.values(item?.data).map((value) => {
+			if (value === null || value === undefined) return '';
```
Comment (YounixM) [inline frontend/src/container/ListOfDashboard/utils.ts:13]: Should we check for all falsy values ?

**Review by YounixM — APPROVED** [review]

**Pull request merged**